source/common/config/well_known_names.cc

```diff
@@ -0,0 +1,86 @@
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Config {
+
+std::vector<std::pair<std::string, std::string>> TagNameValues::getRegexMapping() {
+  std::vector<std::pair<std::string, std::string>> name_regex_pairs;
+
+  // cluster.<cluster_name>.
+  name_regex_pairs.push_back({CLUSTER_NAME, "^cluster\\.((.*?)\\.)"});
+
+  // listener.<port>.
+  name_regex_pairs.push_back({LISTENER_PORT, "^listener\\.((\\d+?)\\.)"});
+
+  // http.<stat_prefix>.
+  name_regex_pairs.push_back({HTTP_CONN_MANAGER_PREFIX, "^http\\.((.*?)\\.)"});
+
+  // vhost.<virtual host name>.
+  name_regex_pairs.push_back({VIRTUAL_HOST, "^vhost\\.((.*?)\\.)"});
+
+  // mongo.<stat_prefix>.
+  name_regex_pairs.push_back({MONGO_PREFIX, "^mongo\\.((.*?)\\.)"});
+
+  // tcp.<stat_prefix>.
+  name_regex_pairs.push_back({TCP_PREFIX, "^tcp\\.((.*?)\\.)"});
+
+  // auth.clientssl.<stat_prefix>.
+  name_regex_pairs.push_back({CLIENTSSL_PREFIX, "^auth\\.clientssl\\.((.*?)\\.)"});
+
+  // ratelimit.<stat_prefix>.
+  name_regex_pairs.push_back({RATELIMIT_PREFIX, "^ratelimit\\.((.*?)\\.)"});
+
+  // http.<stat_prefix>.user_agent.<user_agent>.[base_stat]
+  name_regex_pairs.push_back({HTTP_USER_AGENT, "^http(?=\\.).*?\\.user_agent\\.((.*?)\\.)\\w+?$"});
+
+  // listener.<port>.ssl.cipher.<cipher>
+  name_regex_pairs.push_back({SSL_CIPHER, "^listener(?=\\.).*?\\.ssl\\.cipher(\\.(.*?))$"});
+
+  // mongo.<stat_prefix>.cmd.<cmd>.[base_stat]
+  name_regex_pairs.push_back({MONGO_CMD, "^mongo(?=\\.).*?\\.cmd\\.((.*?)\\.)\\w+?$"});
+
+  // mongo.<stat_prefix>.collection.<collection>.query.[base_stat]
+  name_regex_pairs.push_back(
+      {MONGO_COLLECTION, "^mongo(?=\\.).*?\\.collection\\.((.*?)\\.).*?query.\\w+?$"});
```

Comment:
How come these regexes match all the way to `$`, while the earlier ones don't?

Reply:
Thanks for pointing this out.  I updated the regexes to be more consistent in regards to this.  I added `$` to follow the tags that always precede a base stat at the end of the string.  If the tags are followed by anything more complex, I didn't extend the expression to assume the structure of the more specific part of the expression.  This was done to allow some flexibility for relatively nonspecific tags and keep things a little simpler.